src/app/meetups/meetups.component.ts

```diff
@@ -142,8 +143,58 @@ export class MeetupsComponent implements OnInit, AfterViewInit {
   }
 
   ngOnInit() {
-    this.getMeetups();
+    forkJoin(this.getMeetups(), this.getUserMeetups()).subscribe((results) => {
+      const meetupRes = results[0],
+        userMeetupRes = results[1];
+      this.setupList(meetupRes.rows, userMeetupRes.rows);
+    }, (err) => console.log(err));
     this.meetups.filterPredicate = filterSpecificFields([ 'title', 'description' ]);
   }
 
+  setupList(meetupRes, userMeetupRes) {
+    this.meetups.data = meetupRes.map((m: any) => {
+      const meetup = m.doc || m;
+      const meetupIndex = userMeetupRes.findIndex(usermeetup => {
+        return (meetup._id === usermeetup.doc.meetupId && usermeetup.doc.memberId.indexOf(this.userService.get().name) > -1);
+      });
+      if (meetupIndex > -1) {
+        return { ...meetup, participate: true };
+      }
+      return { ...meetup,  participate: false };
+    });
+  }
+
+  updateMeetup(meetup) {
+    const { _id: meetupId } = meetup;
+    this.router.navigate([ '/meetups/update/' + meetup._id ]);
+  }
+
+  getUserMeetups() {
+    return this.couchService.get('usermeetups/_all_docs?include_docs=true');

```

Comment:
We should find the document that matches the user rather than pulling all docs from CouchDB